### PR TITLE
Refactor app state architecture

### PR DESCRIPTION
**Pull request by nuurcodes**

### CODE REVIEW

File: src/store/config.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      appUrl: 'https://mistX.wtf',
+      contactEmail: 'foo@gmail.com',
+      wethAddress: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+      rpcUrl: 'https://mainnet.infura.io/v3/5505e75053fb4b1ea7ee11fa60f42529',
+      serverUrl: process.env.REACT_APP_SERVER_URL as string,
+      dappId: '6bc363be-f6c0-4593-8edd-761472d54dfe',
+      appName: 'mistX',
+    },
+    [Network.RINKEBY]: {
+      networkId: Network.RINKEBY,
```
Comment (henridf) [inline src/store/config.ts:36]: We're on görli, but i'm not sure what else needs to be set up there... @CodeForcer ?

File: src/hooks/useQuoteSocket.ts [hunk 2]
```diff
@@ -31,55 +32,69 @@ interface IQuoteRequest {
...
-  const [minAmountIn, setMinAmountIn] = useState<number>();
-  const [minAmountOut, setMinAmountOut] = useState<number>();
+  const state = useFormStore();
+
+  const sessionTokenKey = 'SESSION_TOKEN';
+  const sessionToken = localStorage.getItem(sessionTokenKey);
 
   const socket: Socket<QuoteEventsMap, QuoteEventsMap> = io(serverUrl, {
     transports: ['websocket'],
+    auth: { sessionToken },
```
Comment (Dire-0x) [inline src/hooks/useQuoteSocket.ts:54]: I think the token should be passed as `auth.token` not `auth.sessionToken`

File: src/hooks/useQuoteSocket.ts [hunk 3]
```diff
@@ -31,55 +32,69 @@ interface IQuoteRequest {
...
   });
 
-  const debouncedInput: number = useDebounce<number>(amountIn || 0, 5000);
-  const debouncedOutput: number = useDebounce<number>(amountOut || 0, 5000);
+  socket.disconnect();
```
Comment (Dire-0x) [inline src/hooks/useQuoteSocket.ts:57]: What is the reason for the disconnect here?

**Comment by Dire-0x** [comment]
Sockets are also used for transactions and will expand to other uses. It might make more sense to instantiate the socket connection elsewhere and import/pass into `useQuoteSocket`?

File: src/hooks/useQuoteSocket.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+};
+
+/** Hook to interact the with the backend websocket */
+export function useQuoteSocket() {
+  const state = useFormStore();
+
+  const sessionTokenKey = 'SESSION_TOKEN';
+  const sessionToken = localStorage.getItem(sessionTokenKey);
+
+  const socket: Socket<QuoteEventsMap, QuoteEventsMap> = io(serverUrl, {
```
Comment (Dire-0x) [inline src/hooks/useQuoteSocket.ts:52]: Will the socket get instantiated more than once per client?

File: src/store/config.ts (hunk 1 above)
Comment (nuurcodes) [reply src/store/config.ts:36]: I'll probably merge this and change to gorli once we have the Mist token address

File: src/hooks/useQuoteSocket.ts (hunk 3 above)
Comment (nuurcodes) [reply src/hooks/useQuoteSocket.ts]: Had it there for testing, I've removed it now

File: src/hooks/useQuoteSocket.ts (hunk 4 above)
Comment (nuurcodes) [reply src/hooks/useQuoteSocket.ts]: I've refactored based on our call so it shouldn't anymore

**Comment by pygeek** [comment]
I like this 👍🏽

**Pull request merged**